File: kirun-java/src/test/java/com/fincity/nocode/kirun/engine/function/system/math/RandomFloatTest.java
package com.fincity.nocode.kirun.engine.function.system.math;

import static org.junit.jupiter.api.Assertions.assertTrue;

import java.util.Map;

import org.junit.jupiter.api.Test;

import com.fincity.nocode.kirun.engine.repository.reactive.KIRunReactiveFunctionRepository;
import com.fincity.nocode.kirun.engine.repository.reactive.KIRunReactiveSchemaRepository;
import com.fincity.nocode.kirun.engine.runtime.reactive.ReactiveFunctionExecutionParameters;
import com.google.gson.JsonPrimitive;

import reactor.test.StepVerifier;

class RandomFloatTest {

	@Test
	void test() {
		var min = new JsonPrimitive(1.09e2);

		var max = new JsonPrimitive(1000f);

		RandomInt ran = new RandomInt();
		ReactiveFunctionExecutionParameters fep = new ReactiveFunctionExecutionParameters(
				new KIRunReactiveFunctionRepository(), new KIRunReactiveSchemaRepository())
				.setArguments(Map.of("minValue", min, "maxValue", max));

		StepVerifier.create(ran.execute(fep)).expectNextMatches(r -> {
			var x = r.next().getResult().get("value");
			assertTrue(x.getAsFloat() >= min.getAsFloat() && x.getAsFloat() <= max.getAsFloat());
			return true;
		}).verifyComplete();
	}

	@Test
	void test2() {
		var min = new JsonPrimitive(1.09e2);
		RandomInt ran = new RandomInt();
		ReactiveFunctionExecutionParameters fep = new ReactiveFunctionExecutionParameters(
				new KIRunReactiveFunctionRepository(), new KIRunReactiveSchemaRepository())
				.setArguments(Map.of("minValue", min));

		StepVerifier.create(ran.execute(fep)).expectNextMatches(r -> {
			var x = r.next().getResult().get("value");
			assertTrue(x.getAsFloat() >= min.getAsFloat() && x.getAsFloat() <= Float.MAX_VALUE);
			return true;
		}).verifyComplete();
	}

	@Test
	void test3() {
		var max = new JsonPrimitive(1.23e4);
		RandomInt ran = new RandomInt();
		ReactiveFunctionExecutionParameters fep = new ReactiveFunctionExecutionParameters(
				new KIRunReactiveFunctionRepository(), new KIRunReactiveSchemaRepository())
				.setArguments(Map.of("maxValue", max));

		StepVerifier.create(ran.execute(fep)).expectNextMatches(r -> {
			var x = r.next().getResult().get("value");
			assertTrue(x.getAsFloat() >= Float.MIN_VALUE && x.getAsFloat() <= max.getAsFloat());
			return true;
		}).verifyComplete();
	}
}

File: kirun-js/__tests__/engine/function/system/math/RandomFloatTest.ts
import { FunctionExecutionParameters } from '../../../../../src/engine/runtime/FunctionExecutionParameters';
import { KIRunFunctionRepository, KIRunSchemaRepository, Namespaces } from '../../../../../src';
import { MathFunctionRepository } from '../../../../../src/engine/function/system/math/MathFunctionRepository';

test('rand Float 1', async () => {
    const rand = (await new MathFunctionRepository().find(Namespaces.MATH, 'RandomFloat'))!;
    let min = 1,
        max = 10;

    let fep: FunctionExecutionParameters = new FunctionExecutionParameters(
        new KIRunFunctionRepository(),
        new KIRunSchemaRepository(),
    ).setArguments(
        new Map([
            ['minValue', min],
            ['maxValue', max],
        ]),
    );

    let num: number = (await rand.execute(fep)).allResults()[0].getResult().get('value');

    expect(num).toBeLessThanOrEqual(max);
    expect(num).toBeGreaterThanOrEqual(min);
});

test('rand Float 2', async () => {
    const rand = (await new MathFunctionRepository().find(Namespaces.MATH, 'RandomFloat'))!;
    let min = 0.1,
        max = 0.9;

    let fep: FunctionExecutionParameters = new FunctionExecutionParameters(
        new KIRunFunctionRepository(),
        new KIRunSchemaRepository(),
    ).setArguments(
        new Map([
            ['minValue', min],
            ['maxValue', max],
        ]),
    );

    let num: number = (await rand.execute(fep)).allResults()[0].getResult().get('value');

    expect(num).toBeLessThanOrEqual(max);
    expect(num).toBeGreaterThanOrEqual(min);
});

test('rand Float 3', async () => {
    const rand = (await new MathFunctionRepository().find(Namespaces.MATH, 'RandomFloat'))!;
    let min = 1,
        max = 2147483647;

    let fep: FunctionExecutionParameters = new FunctionExecutionParameters(
        new KIRunFunctionRepository(),
        new KIRunSchemaRepository(),
    ).setArguments(new Map([['minValue', min]]));

    let num: number = (await rand.execute(fep)).allResults()[0].getResult().get('value');

    expect(num).toBeLessThanOrEqual(max);
    expect(num).toBeGreaterThanOrEqual(min);
});

test('rand Float 4', async () => {
    const rand = (await new MathFunctionRepository().find(Namespaces.MATH, 'RandomFloat'))!;
    let min = 1.1,
        max = 1.2;

    let fep: FunctionExecutionParameters = new FunctionExecutionParameters(
        new KIRunFunctionRepository(),
        new KIRunSchemaRepository(),
    ).setArguments(
        new Map([
            ['minValue', min],
            ['maxValue', max],
        ]),
    );

    let num: number = (await rand.execute(fep)).allResults()[0].getResult().get('value');

    expect(num).toBeLessThanOrEqual(max);
    expect(num).toBeGreaterThanOrEqual(min);
});
